src/client/ducks/repositories/selectors/index.js

```diff
@@ -72,3 +73,27 @@ export const getRepositoryId = createSelector(
     _.findKey(entities, (v) => (v.fullName === repoName))
 )
 
+export const getChangesetList = (state: Object) => _.values(state.entities.changesets)
+
+export const getChangesetsWithAvatars = (state: Object) => createSelector(
+  getChangesetList, userEntitiesSelector,
+  (changesets, users) =>
+    _.values(changesets.map(changeset => {
+      const user = _.values(users).find(x => x.username === changeset.author)
```

Comment:
this name comparing is not very elegant. It'd be more convenient if the `author` specified in GraphQL query was `UserType` (and in `ChangesetType` it is indeed. Alas, it's most probably specified as string on the server's side somewhere). Then querying for all users could be avoided as well. But maybe there is a better way here that I can't see now.